op-batcher/batcher/driver.go

```diff
@@ -945,3 +919,136 @@ func logFields(xs ...any) (fs []any) {
 	}
 	return fs
 }
+
+type ChannelStatuser interface {
+	isFullySubmitted() bool
+	isTimedOut() bool
+	LatestL2() eth.BlockID
+	MaxInclusionBlock() uint64
+}
+
+type SyncActions struct {
+	clearState      *eth.BlockID
+	blocksToPrune   int
+	channelsToPrune int
+	waitForNodeSync bool
+	blocksToLoad    [2]uint64 // the range [start,end] that should be loaded into the local state.
+	// NOTE this range is inclusive on both ends, which is a change to previous behaviour.
+}
```

Comment:
From chat with @sebastianst today, perhaps we could change the API so that `waitForNodeSync` and `clearState` could be extracted into two different error values returned This might make the data structure match the different actions better, and make it impossible to do incorrect things. 

Or, going in the other direction, should we _always_ wait for node sync before taking any action?

Reply:
It might be better for the error to be "needs node sync" or "sequencer out of sync". The caller can then decide how to deal with that error, e.g. either:
1. Wait another ticker
2. or wait _now_ for the sequencer to catch up

Reply:
Made the "needs sync" case an error here [47578b2](https://github.com/ethereum-optimism/optimism/pull/13060/commits/47578b23642403a90702663723751495854e4b58)

Reply:
Also, `clearState` does need to specify a value (a pointer in fact) which isn't just a boolean, so it seems less natural to make it an error.